File: app/src/main/java/dagger/nambv/app/demo/view/fragment/PromotionsFragmentComponent.java
package dagger.nambv.app.demo.view.fragment;

import dagger.Subcomponent;
import dagger.nambv.app.demo.di.scope.ActivityScope;

@ActivityScope
@Subcomponent(modules = {PromotionsFragmentModule.class})
public interface PromotionsFragmentComponent {
    void inject(PromotionsFragment promotionsFragment);
}

File: app/src/main/java/dagger/nambv/app/demo/view/fragment/PromotionsFragmentComponent.kt
package dagger.nambv.app.demo.view.fragment

import dagger.Subcomponent
import dagger.nambv.app.demo.di.scope.ActivityScope

@ActivityScope
@Subcomponent(modules = arrayOf(PromotionsFragmentModule::class))
interface PromotionsFragmentComponent {
    fun inject(promotionsFragment: PromotionsFragment)
}
